src/javaRosa.js

```diff
@@ -576,7 +576,12 @@ define([
 
         widget.save = function () {
             // override save to call out to rename itext
-            widget.mug.setItextId(widget.path, widget.getValue());
+            if (widget.mug.setItextId(widget.path, widget.getValue())) {
```

Comment:
Coming from the Python world, this is generally discouraged. One reason being if someone adds another condition using `||` (or) or `&&` (and) it can cause unexpected things to happen due to short-circuit logical evaluation. For example (silly example):
```js
if (true && widget.mug.setItextId(widget.path, widget.getValue())) {
    // we're here, but setItextId did't get called
}
```
My rule of thumb is if you really want that function to be called then you should assign it's return value to a variable and test the value of that. If it doesn't matter if it gets called, then testing it in a condition is probably fine.